src/grpc.rs

```diff
@@ -75,3 +86,158 @@ impl ClientPolicies for Server {
         })))
     }
 }
+
+fn to_client_policy(svc: &OutboundService) -> Result<proto::ClientPolicy, tonic::Status> {
+    let http_routes = svc
+        .http_routes
+        .iter()
+        .map(|(route_ref, route)| to_http_route(&route_ref, route))
+        .collect::<Result<_, _>>()?;
+
+    Ok(proto::ClientPolicy {
+        fully_qualified_name: svc.fqdn.as_ref().map(|s| s.to_string()).unwrap_or_default(),
+        endpoint: None,
+        protocol: Some(ProxyProtocol {
+            kind: Some(Kind::Detect(Detect {
+                timeout: Default::default(),
+                http_routes,
+            })),
+        }),
+        filters: to_filters(&svc.client_policies)?,
+    })
+}
+
+fn to_http_route(
+    reference: &InboundHttpRouteRef,
+    OutboundHttpRoute {
+        hostnames,
+        client_policies,
+        rules,
+        creation_timestamp: _,
+    }: &OutboundHttpRoute,
+) -> Result<proto::HttpRoute, tonic::Status> {
+    let metadata = meta::Metadata {
+        kind: Some(match reference {
+            InboundHttpRouteRef::Default(name) => meta::metadata::Kind::Default(name.to_string()),
+            InboundHttpRouteRef::Linkerd(name) => meta::metadata::Kind::Resource(meta::Resource {
+                group: "policy.linkerd.io".to_string(),
+                kind: "HTTPRoute".to_string(),
+                name: name.to_string(),
+            }),
+        }),
+    };
+
+    let hosts = hostnames.into_iter().map(convert_host_match).collect();
+
+    let rules = rules
+        .into_iter()
+        .map(|rule| convert_rule(&client_policies, rule))
+        .collect::<Result<_, _>>()?;
+
+    Ok(proto::HttpRoute {
+        metadata: Some(metadata),
+        hosts,
+        rules,
+    })
+}
+
+fn convert_host_match(h: &HostMatch) -> http_route::HostMatch {
+    http_route::HostMatch {
+        r#match: Some(match h {
+            HostMatch::Exact(host) => http_route::host_match::Match::Exact(host.clone()),
+            HostMatch::Suffix { reverse_labels } => {
+                http_route::host_match::Match::Suffix(http_route::host_match::Suffix {
+                    reverse_labels: reverse_labels.to_vec(),
+                })
+            }
+        }),
+    }
+}
+
+fn to_filters(policies: &PolicySet) -> Result<Vec<proto::Filter>, tonic::Status> {
+    let filters = policies
+        .policies
+        .values()
+        .flat_map(|spec| &spec.filters)
+        .map(convert_filter)
+        .collect::<Result<_, _>>()?;
+    Ok(filters)
+}
```

Comment:
i don't think this is quite right. the `PolicySet` on the `OutboundService`'s `policies` field contains _all_ the ClientPolicies that apply to that service. The `bindings` field on the `PolicySet`, on the other hand, contains the policies that are bound by `ClientPolicyBinding`s, including the pod selector. This is why the `async-stream` we return from the API service holds a watch on the pod metadata, so that we can select only the policies that are bound to that client.

i believe this function should look more like this:
```suggestion
fn to_filters(policies: &PolicySet, pod: &pod::Meta) -> Result<Vec<proto::Filter>, tonic::Status> {
    let filters = policies
        .bindings
        .values()
        .filter(|binding| binding.client_pod_selector.matches(pod.labels))
        .flat_map(|binding| binding.policies.iter().flat_map(|spec| spec.filters.iter()))
        .map(convert_filter)
        .collect::<Result<_, _>>()?;
    Ok(filters)
}
```

(and, of course, we'd have to update the callers of this function to thread through the pod metadata)